tests/MongoDB.Driver.Core.Tests/Core/Compression/CompressedMessageBinaryEncoderTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Core.Configuration;
+using MongoDB.Driver.Core.WireProtocol.Messages;
+using MongoDB.Driver.Core.WireProtocol.Messages.Encoders;
+using MongoDB.Driver.Core.WireProtocol.Messages.Encoders.BinaryEncoders;
+using Moq;
+using Xunit;
+
+namespace MongoDB.Driver.Core.Compression
+{
+    public class CompressedMessageBinaryEncoderTests
+    {
+        private readonly MessageEncoderSettings _messageEncoderSettings = new MessageEncoderSettings();
+
+        [Fact]
+        public void Constructor_should_throw_the_exception_if_the_stream_is_null()
+        {
+            Record.Exception(() => CreateSubject(null))
```

Comment:
This complex logical one liner is blending the act/assert sections. If you look at other tests this normally would have been written as:

            var exception = Record.Exception(() => CreateSubject(null));

            var e = exception.Should().BeOfType<ArgumentNullException>().Subject;
            e.ParamName.Should().Be("stream");